zanata-war/src/main/webapp/WEB-INF/layout/version/upload_translation_modal.xhtml

```diff
@@ -4,63 +4,87 @@
   xmlns:zanata="http://java.sun.com/jsf/composite/zanata"
   xmlns:a4j="http://richfaces.org/a4j"
   xmlns:rich="http://richfaces.org/rich"
+  xmlns:f="http://java.sun.com/jsf/core"
   class="modal" id="uploadTranslationDialog" tabindex="-1" role="dialog"
   aria-labelledby="myModalLabel" aria-hidden="true">
 
   <script>
     $(document).ready(function() {
-      var uploadButton = getUploadButton();
       zanata.loader.deactivate('#upload-translation-loader');
-      jQuery(uploadButton).attr('disabled', true);
-      jQuery(uploadButton).removeClass('is-hidden');
-
-      $(getUploadFile()).change(function() {
-        var uploadButton = getUploadButton();
-        if(jQuery(getUploadFile()).val() != '') {
-          jQuery(uploadButton).removeAttr('disabled');
-        } else {
-          jQuery(uploadButton).attr('disabled', true);
+      toggleUploadTranslationDocButton(false, true);
+      $(getUploadFileComponent()).change(function() {
+        var isFileSelected = getUploadFileName() != '';
+        if(isFileSelected) {
+          renderAdapterSelection(getUploadFileName());
         }
+        toggleUploadTranslationDocButton(isFileSelected, true);
       });
     });
 
-    function getUploadButton() {
+    function toggleUploadTranslationDocButton(enable, display) {
+      var uploadButton = getUploadTranslationDocButton();
+      jQuery(uploadButton).prop('disabled', !enable);
+      jQuery(uploadButton).toggleClass('is-hidden', !display);
+    }
```

Comment:
- [ ] This function is confusing at the call site. I think it should be split into `setUploadTranslationDocButtonEnabled(boolean)` and `SetUploadTranslationDocButtonVisible(boolean)` so that it is clear at the call site what is happening.